airbyte-integrations/connectors/destination-timeplus/integration_tests/integration_test.py

```diff
@@ -0,0 +1,63 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+import json
+from typing import Any, Mapping
+import pytest
+from airbyte_cdk import AirbyteLogger
+from airbyte_cdk.models import AirbyteMessage, AirbyteRecordMessage, Type, AirbyteStream, ConfiguredAirbyteCatalog, ConfiguredAirbyteStream, DestinationSyncMode, Status, SyncMode
+from destination_timeplus import DestinationTimeplus
+
+@pytest.fixture(name="config")
+def config_fixture() -> Mapping[str, Any]:
+    with open("secrets/config.json", "r") as f:
+        return json.loads(f.read())
+
+@pytest.fixture(name="configured_catalog")
+def configured_catalog_fixture() -> ConfiguredAirbyteCatalog:
+    stream_schema = {"type": "object", "properties": {"string_col": {"type": "str"}, "int_col": {"type": "integer"}}}
+
+    append_stream = ConfiguredAirbyteStream(
+        stream=AirbyteStream(name="append_stream", json_schema=stream_schema, supported_sync_modes=[SyncMode.incremental]),
+        sync_mode=SyncMode.incremental,
+        destination_sync_mode=DestinationSyncMode.append,
+    )
+
+    overwrite_stream = ConfiguredAirbyteStream(
+        stream=AirbyteStream(name="overwrite_stream", json_schema=stream_schema, supported_sync_modes=[SyncMode.incremental]),
+        sync_mode=SyncMode.incremental,
+        destination_sync_mode=DestinationSyncMode.overwrite,
+    )
+
+    return ConfiguredAirbyteCatalog(streams=[append_stream, overwrite_stream])      
+
+def test_check_valid_config(config: Mapping):
+    outcome = DestinationTimeplus().check(AirbyteLogger(), config)
+    assert outcome.status == Status.SUCCEEDED
+
+
+def test_check_invalid_config():
+    outcome = DestinationTimeplus().check(AirbyteLogger(), {"secret_key": "not_a_real_secret"})
+    assert outcome.status == Status.FAILED  
+
+def test_write(config: Mapping):
+    test_schema = {"type": "object", "properties": {"str_col": {"type": "str"}, "int_col": {"type": "integer"}}}
+
+    test_stream = ConfiguredAirbyteStream(
+        stream=AirbyteStream(name="test_stream", json_schema=test_schema, supported_sync_modes=[SyncMode.incremental]),
+        sync_mode=SyncMode.incremental,
+        destination_sync_mode=DestinationSyncMode.append,
+    )
+
+    records = [AirbyteMessage(
+        type=Type.RECORD, record=AirbyteRecordMessage(stream="test_stream", data={
+            "str_col": "example",
+            "int_col": 1,
+        }, emitted_at=0)
+    )]
+    dest = DestinationTimeplus()
+    dest.write(
+        config=config, 
+        configured_catalog=test_stream,
+        input_messages=records
+    )
```

Comment:
Sorry the delay @jovezhong I added this function to test the write function. I ran it and it's not sending data to Timeplus. Please make sure it works.

Reply:
Also I tried the connector locally and it's not working.
![image](https://github.com/airbytehq/airbyte/assets/5154322/e67898ba-0e12-42e4-8d6a-f5c70c9729ed)

Reply:
I will be offline for 1 hour, then join a meeting.

Take a quick look at the error message. It is trying to add an event in Timeplus for `campaigns` stream for `lastProcessingFinishedAt` column, but this column is not defined. There is only a `lastUpdatedTime` column. Not sure whether hubspot will produce event with extra column which is not defined in the schema.

I will take a look later today after other meetings. In the meanwhile, you may try other sources such as CSV.

I have a daily job to push Hubspot to Timeplus via this airbyte connector, but mainly for contacts/companies, not campaigns

Reply:
Anything please ping me on Slack.